packages/react/src/components/ComposedModal/ComposedModal.tsx

```diff
@@ -0,0 +1,457 @@
+import React, {
+  useRef,
+  useEffect,
+  useState,
+  type MouseEvent,
+  type KeyboardEvent,
+  type HTMLAttributes,
+  type ReactNode,
+  type ReactElement,
+} from 'react';
+import { isElement } from 'react-is';
+import PropTypes from 'prop-types';
+import { ModalHeader, type ModalHeaderProps } from './ModalHeader';
+import { ModalFooter, type ModalFooterProps } from './ModalFooter';
+
+import cx from 'classnames';
+
+import toggleClass from '../../tools/toggleClass';
+import requiredIfGivenPropIsTruthy from '../../prop-types/requiredIfGivenPropIsTruthy';
+
+import wrapFocus from '../../internal/wrapFocus';
+import { usePrefix } from '../../internal/usePrefix';
+
+export interface ModalBodyProps extends HTMLAttributes<HTMLDivElement> {
+  /** Specify the content to be placed in the ModalBody. */
+  children?: ReactNode;
+
+  /**
+   * Provide whether the modal content has a form element.
+   * If `true` is used here, non-form child content should have `bx--modal-content__regular-content` class.
+   */
+  hasForm?: boolean;
+
+  /**
+   * Specify whether the modal contains scrolling content
+   */
+  hasScrollingContent?: boolean;
+}
+
+export const ModalBody = React.forwardRef<HTMLDivElement, ModalBodyProps>(
+  function ModalBody(
+    {
+      className: customClassName,
+      children,
+      hasForm,
+      hasScrollingContent,
+      ...rest
+    },
+    ref
+  ) {
+    const prefix = usePrefix();
+    const contentClass = cx(
+      `${prefix}--modal-content`,
+      hasForm && `${prefix}--modal-content--with-form`,
+      hasScrollingContent && `${prefix}--modal-scroll-content`,
+      customClassName
+    );
```

Comment:
we've been using this className standard to prevent undefined classes 
<img width="722" alt="image" src="https://user-images.githubusercontent.com/40550942/233187280-b6854272-456c-4ac2-9987-e22f4bdd7bc9.png">